### PR TITLE
WIP: Upload Photo FE #35

### PR DESCRIPTION
**Pull request by FidaaHersh**
References issue #43

### CODE REVIEW

File: server/controllers/uploadPhoto.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    Bucket: process.env.S3_BUCKET,
+    ContentType: type.mime,
+    Key: `${name}.${type.ext}`,
+  };
+  return s3.upload(params).promise();
+};
+
+module.exports = uploadPhoto = (req, res) => {
+  console.log(req);
+  const form = new multiparty.Form();
```
Comment (Hadeel-Salamin) [inline server/controllers/uploadPhoto.js]: no need to throw new error?
e.g
``` 
      return res.json({error: "something wrong"})
```

File: client/src/App.js [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 import { Route, BrowserRouter as Router } from 'react-router-dom';
 import GetDetails from './components/GetDetails';
+import Upload from './components/Upload';
 
 function App() {
   return (
     <Router>
-      <Route path="/GetDetails" component={GetDetails} />
+      <div className="App">
+        <Route path="/GetDetails" component={GetDetails} />
```
Comment (jema28) [inline client/src/App.js]: You need to keep your path's consistent. They should all be kebab case, like `upload-photo`.

File: server/controllers/uploadPhoto.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  form.parse(req, async (error, fields, files) => {
+    if (error) throw new Error(error);
```
Comment (Hadeel-Salamin) [inline server/controllers/uploadPhoto.js]: no need to throw new error?
e.g
```
return res.json({error: "something wrong"})
```

File: client/src/components/Upload/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react';
+import { Route, BrowserRouter as Router } from 'react-router-dom';
```
Comment (jema28) [inline client/src/components/Upload/index.js]: The router isn't needed here.

Screenshot 2019-03-24 at 10 32 34

File: client/src/components/Upload/index.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+class Upload extends Component {
+  state = {
+    loc: null,
+  };
+  handleUploadFile = event => {
+    const data = new FormData();
+    data.append('file', event.target.files[0]);
+    axios.post('/add-to-amazon', data).then(response => {
+      console.log(response);
```
Comment (jema28) [inline client/src/components/Upload/index.js]: Delete this console log.

File: server/controllers/uploadPhoto.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    Body: buffer,
+    Bucket: process.env.S3_BUCKET,
+    ContentType: type.mime,
+    Key: `${name}.${type.ext}`,
+  };
+  return s3.upload(params).promise();
+};
+
+module.exports = uploadPhoto = (req, res) => {
+  console.log(req);
```
Comment (jema28) [inline server/controllers/uploadPhoto.js]: Please delete this console.log.

File: server/controllers/uploadPhoto.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  form.parse(req, async (error, fields, files) => {
+    if (error) throw new Error(error);
+    try {
+      const { path } = files.file[0];
+      const buffer = fs.readFileSync(path);
+      const type = fileType(buffer);
+      const timestamp = Date.now().toString();
+      const fileName = `bucketFolder/${timestamp}-lg`;
+      const data = await uploadFile(buffer, fileName, type);
+      console.log(res);
```
Comment (jema28) [inline server/controllers/uploadPhoto.js]: ^ And this one.

**Review by jema28 — CHANGES_REQUESTED** [review]

**Review by amanshawar — COMMENTED** [review]
small things
you write upload-photo insted of uploadPhoto 
        <Route path="/upload-photo" component={Upload} />
you can delete uploadPhoto 
module.exports = uploadPhoto = (req, res) => {
delete logs 

just good job

**Comment by jema28** [comment]
@amanshawar The paths should be in kebab case so `upload-photo` is correct. All should follow this same style.

**Review by Hadeel-Salamin — APPROVED** [review]

**Review by jema28 — APPROVED** [review]
💯

**Pull request merged**